Replace _countof with COUNTOF

ton opened the pull request:
Instead of depending on the MSVC specific `_countof` macro to determine
the number of elements in an array, use `std::extent`.

mhalber commented:
Not in love with using std for this little project. If portability is a concern we can define standard COUNTOF macro.

ton commented:
Of course, any solution that makes code compile under GCC will be an improvement for me. I would need to revisit this again to see how `std::extent` can be avoided.

ton committed:
Replace _countof with custom macro

Instead of depending on the MSVC's specific `_countof` macro to
determine the number of elements in an array, define `COUNTOF` and use
that instead to improve portability.
